client/engine/chainservice/eth_chainservice.go

```diff
@@ -39,7 +48,13 @@ func NewEthChainService(na *NitroAdjudicator.NitroAdjudicator, naAddress common.
 	return ecs
 }
 
-func (cc EthChainService) In() chan<- protocols.ChainTransaction {
+func (cc *EthChainService) Input(tx protocols.ChainTransaction) chan struct{} {
```

Comment:
Do you have a vision for the `In` vs `Input` APIs?

Reply:
Not really, very open to suggestions. The main thing was to not expose the input `chan` directly. By wrapping in a function call, that function can `make` the `done` chan and wrap it around the `tx` (saving the consumer from having to do that).

Reply:
Thanks for the explanation! Do you have any background on the design decision for the chain service to accept inputs via a channel (as opposed to a function call)? Specifically, I am thinking of [this API](https://github.com/statechannels/go-nitro/blob/42e25a802693f8daea10f5b3b5e5638e6d721082/client/engine/chainservice/simplechainservice.go#L38).

Reply:
That's a good question. That decision was made very early on, and there may not be a strong justification for it. I think probably the original idea was loosely based on transaction submission being an async process.

But now we come to revisit the design, and it seems that we would _like_ the consumer to _block_ until the transaction is submitted (e.g. `NitroAdjudicator.Deposit()` is called). So this could be an argument for not using channels at all.

Reply:
Or at least we would like the consumer to be able to block if it wants to. The current system on `main` is really "fire and forget".